### ISSUE TITLE
mysqli methods have wrong parameter counts

### ISSUE THREAD

**Issue by jaydiablo** [vimeo/psalm#669]
Ran into a case where psalm was complaining about some methods in a class that extends PHP's `mysqli` class, something like this perhaps:

```php
class db extends mysqli 
{
    public function close() // claims too few parameters http://php.net/manual/en/mysqli.close.php
    {
    }

    public function prepare($sql) // claims too few parameters http://php.net/manual/en/mysqli.prepare.php
    {
    }

    public function commit($flags = null, $name = null) // claims first is wrong type, too many params http://php.net/manual/en/mysqli.commit.php
    {
    }

    public function real_escape_string($string) // claims too few parameters http://php.net/manual/en/mysqli.real-escape-string.php
    {
    }
}
```

There may be others, I didn't audit all of them. It seems the issue stems from psalm's definitions of these methods being copies of the procedural style functions, which have different parameter requirements (usually take the `mysqli` resource as the first parameter).

**Comment by jaydiablo** [comment]
Another case where `mysqli::prepare` isn't marked that it may return `false` (or `mysqli_stmt`).

http://php.net/manual/en/mysqli.prepare.php

>mysqli_prepare() returns a statement object or FALSE if an error occurred.

**Comment by muglug** [comment]
Thanks! Should be fixed now.

**Comment by weirdan** [comment]
@muglug there's several more link arguments to be dropped. Also, there's quite a few **properties** of `mysqli` class in the CallMap, and I don't think they should be there. Or should they?
